### PR TITLE
File upload

### PR DESCRIPTION
**Pull request by ma12-co**
Signed-off-by: Marco Ambrosini <marcoambrosini@pm.me>

### CODE REVIEW

File: src/components/NewMessageForm/NewMessageForm.vue [hunk 1]
```diff
@@ -22,16 +22,39 @@
...
+							icon="icon-upload"
+							@click.prevent="clickImportInput">
+							Upload from device
+						</ActionButton>
+						<ActionButton
+							v-if="!currentUserIsGuest"
+							:close-after-click="true"
+							icon="icon-folder"
+							@click.prevent="handleFileShare">
+							Add from Nextcloud
```
Comment (nickvergessen) [inline src/components/NewMessageForm/NewMessageForm.vue]:
```suggestion
							{{ t('spreed', 'Share from Files') }}
```

File: src/components/NewMessageForm/NewMessageForm.vue [hunk 2]
```diff
@@ -22,16 +22,39 @@
...
+						default-icon="icon-clip-add-file"
+						class="new-message-form__button">
+						<ActionText>Add a file</ActionText>
+
+						<ActionButton
+							v-if="!currentUserIsGuest"
+							:close-after-click="true"
+							icon="icon-upload"
+							@click.prevent="clickImportInput">
+							Upload from device
```
Comment (nickvergessen) [inline src/components/NewMessageForm/NewMessageForm.vue]:
```suggestion
							{{ t('spreed', 'Upload new file') }}
```

File: src/components/NewMessageForm/NewMessageForm.vue [hunk 3]
```diff
@@ -22,16 +22,39 @@
...
 				<div
 					class="new-message-form__button">
-					<button
-						v-if="!currentUserIsGuest"
-						class="new-message-form__button icon-clip-add-file"
-						@click.prevent="handleFileShare" />
+					<Actions
+						default-icon="icon-clip-add-file"
+						class="new-message-form__button">
+						<ActionText>Add a file</ActionText>
```
Comment (nickvergessen) [inline src/components/NewMessageForm/NewMessageForm.vue]:
```suggestion
						<ActionText>{{ t('spreed', 'Add file') }}</ActionText>
```

File: src/components/NewMessageForm/NewMessageForm.vue [hunk 4]
```diff
@@ -251,6 +281,25 @@ export default {
...
+		async processFiles(event) {
+			// Get the files array
+			const files = Object.values(event.target.files)
+			// Copy each file into the remote path
+			for (let i = 0; i < files.length; i++) {
+				const userId = this.$store.getters.getUserId()
+				try {
+					await client.putFileContents(`/files/${userId}/` + files[i].name, files[i], { overwrite: false })
+				} catch (exception) {
+					console.debug('Error while uploading file' + exception)
```
Comment (nickvergessen) [inline src/components/NewMessageForm/NewMessageForm.vue]: Please add a toast message!

File: src/components/NewMessageForm/NewMessageForm.vue [hunk 5]
```diff
@@ -251,6 +281,25 @@ export default {
...
+		},
+		// Uploads the files to the root files directory
+		async processFiles(event) {
+			// Get the files array
+			const files = Object.values(event.target.files)
+			// Copy each file into the remote path
+			for (let i = 0; i < files.length; i++) {
+				const userId = this.$store.getters.getUserId()
+				try {
+					await client.putFileContents(`/files/${userId}/` + files[i].name, files[i], { overwrite: false })
```
Comment (nickvergessen) [inline src/components/NewMessageForm/NewMessageForm.vue]: after upload, the path should be shared :laughing: 
So I guess it's time to split the API call from the filepicker in `handleFileShare` and reuse it in a shared function

File: src/components/NewMessageForm/NewMessageForm.vue (hunk 4 above)
Comment (nickvergessen) [reply src/components/NewMessageForm/NewMessageForm.vue]: `import { showError } from '@nextcloud/dialogs'` should be used

**Comment by szaimen** [comment]
🎉

**Comment by mat-l** [comment]
Would love to see this feature in the mobile apps. 😍😎

**Pull request merged**